src/models/anthropic/messages.ts

```diff
@@ -0,0 +1,430 @@
+   * deterministic.
+   *
+   * @default 1.0
+   */
+  @omitif("this.temperature == 1.0")
+  temperature: f64 = 1.0;
+
+  /**
+   * How the model should use the provided tools.
+   */
+  @alias("tool_choice")
+  @omitnull()
+  toolChoice: ToolChoice | null = null;
+
+  /**
+   * Definitions of tools that the model may use.
+   *
+   * Tools can be used for workflows that include running client-side tools and
+   * functions, or more generally whenever you want the model to produce a particular
+   * JSON structure of output.
+   *
+   * See Anthropic's [guide](https://docs.anthropic.com/en/docs/tool-use) for more details.
+   */
+  @omitnull()
+  tools: Tool[] | null = null;
+
+  /**
+   * Only sample from the top K options for each subsequent token.
+   *
+   * Recommended for advanced use cases only. You usually only need to use
+   * `temperature`.
+   */
+  // TODO: decide whether to include this field
+  // @alias("top_k")
+  // @omitnull()
+  // topK: i64 | null = null;
+
+  /**
+   * Use nucleus sampling.
+   *
+   * You should either alter `temperature` or `top_p`, but not both.
+   *
+   * Recommended for advanced use cases only. You usually only need to use
+   * `temperature`.
+   */
+  @alias("top_p")
+  @omitif("this.topP == 0.999")
+  topP: f64 = 0.999;
+}
+
+
+@json
+class Metadata {
+  /**
+   * An external identifier for the user who is associated with the request.
+   *
+   * This should be a uuid, hash value, or other opaque identifier. Anthropic may use
+   * this id to help detect abuse. Do not include any identifying information such as
+   * name, email address, or phone number.
+   */
+  @alias("user_id")
+  userId: string | null = null;
+}
+
+/**
+ * A tool object that the model may call.
+ */
+@json
+export class Tool {
+  /**
+   * Name of the tool.
+   */
+  name!: string;
+
+  /**
+   * Optional, but strongly-recommended description of the tool.
+   */
+  @omitnull()
+  description: string | null = null;
+
+  /**
+   * [JSON schema](https://json-schema.org/) for this tool's input.
+   *
+   * This defines the shape of the `input` that your tool accepts and that the model
+   * will produce.
+   */
+  @alias("input_schema")
+  inputSchema!: InputSchema;
+}
+
+
+@json
+abstract class ToolChoice {
+  type!: string;
+
+  constructor(type: string) {
+    this.type = type;
+  }
+}
+
+/**
+ * The model will automatically decide whether to use tools.
+ */
+@json
+export class ToolChoiceAuto extends ToolChoice {
+  constructor() {
+    super("auto");
+  }
+}
+
+/**
+ * The model will use any available tools.
+ */
+@json
+export class ToolChoiceAny extends ToolChoice {
+  constructor() {
+    super("any");
+  }
+}
+
+/**
+ * The model will use the specified tool with `tool_choice.name`.
+ */
+@json
+export class ToolChoiceTool extends ToolChoice {
+  /**
+   * The name of the tool to use.
+   */
+  name!: string;
+
+  constructor() {
+    super("tool");
+  }
+}
+
+
+@json
+class Properties {
+  [key: string]: any;
```

Comment:
Can't use `any` in AssemblyScript.  But also, I think this class and also the `InputSchema` class should be removed.  The input schema is an embedded JSON Schema document - which could have anything that JSON Schema allows.  We need to find some way to have a field be dynamic enough to handle this.  I'll explore some ideas.

Reply:
BTW, we have the same problem in OpenAI.  I just stubbed a nullable string for now, but that's not really sufficient.

https://github.com/hypermodeAI/models-as/blob/8c6e6994096321ae032f9001d9b6aaf350d47941/src/models/openai/chat.ts#L362-L369

Reply:
We'll use this, when ready:
https://github.com/JairusSW/as-json/issues/88

Reply:
got it, i've stubbed it with a nullable string as well, and added a TODO.

Reply:
i stubbed `ToolUseBlock.input` as well for the same reason, it's an arbitrary json in the API response.